language/move-lang/src/expansion/ast.rs

```diff
@@ -215,6 +215,11 @@ pub type LValue = Spanned<LValue_>;
 pub type LValueList_ = Vec<LValue>;
 pub type LValueList = Spanned<LValueList_>;
 
+pub type LValueWithRange_ = (LValue, Exp);
```

Comment:
If that's ok, I'm going to rename `LValueWithRange` into `ConstrainedLValue`. (I realize now that number range is just a special case, next to type domain and vectors)

Reply:
Looking at the [wikipedia page for Quantifiers](https://en.wikipedia.org/wiki/Quantifier_(logic)), the two appropriate expressions seem to be "domain of discourse" and "range of quantification".

Reply:
I guess the current name is fine.